iot_config/src/session_key_service.rs

```diff
@@ -1,43 +1,175 @@
-use crate::{GrpcResult, GrpcStreamRequest, GrpcStreamResult};
+use crate::{
+    session_key::{self, SessionKeyFilter},
+    GrpcResult, GrpcStreamRequest, GrpcStreamResult, Settings,
+};
+use anyhow::Result;
+use file_store::traits::MsgVerify;
+use futures::stream::StreamExt;
+use helium_crypto::PublicKey;
 use helium_proto::services::iot_config::{
-    self, SessionKeyFilterGetReqV1, SessionKeyFilterListReqV1, SessionKeyFilterStreamReqV1,
-    SessionKeyFilterStreamResV1, SessionKeyFilterUpdateReqV1, SessionKeyFilterUpdateResV1,
-    SessionKeyFilterV1,
+    self, ActionV1, SessionKeyFilterGetReqV1, SessionKeyFilterListReqV1,
+    SessionKeyFilterStreamReqV1, SessionKeyFilterStreamResV1, SessionKeyFilterUpdateReqV1,
+    SessionKeyFilterUpdateResV1, SessionKeyFilterV1,
+};
+use sqlx::{Pool, Postgres};
+use tokio::{
+    pin,
+    sync::broadcast::{Receiver, Sender},
 };
-use tonic::Request;
+use tonic::{Request, Response, Status};
+
+pub struct SessionKeyFilterService {
+    admin_pubkey: PublicKey,
+    pool: Pool<Postgres>,
+    update_channel: Sender<SessionKeyFilterStreamResV1>,
+}
+
+impl SessionKeyFilterService {
+    pub async fn new(settings: &Settings) -> Result<Self> {
+        let (update_tx, _) = tokio::sync::broadcast::channel(128);
+        Ok(Self {
+            admin_pubkey: settings.admin_pubkey()?,
+            pool: settings.database.connect(10).await?,
```

Comment:
nit: add a const for the connection pool size

Reply:
@bbalser is making changes to support postgres password vs. iam role-based authentication and i believe his change is going to include this type of centralized pool configuration